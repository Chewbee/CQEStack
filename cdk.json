{
  "app": "npx ts-node bin/CQE.ts"
}
