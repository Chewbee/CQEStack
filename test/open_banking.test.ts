import { expect as expectCDK, matchTemplate, MatchStyle, haveResource, SynthUtils } from '@aws-cdk/assert'
import cdk = require('@aws-cdk/core');
import OpenBanking = require('../lib/open_banking-stack');

test('Stack built', () => {
  const app = new cdk.App()
  // WHEN
  const stack = new OpenBanking.OpenBankingStack(app, 'MyTestStack')
  // THEN
  return undefined
  // expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
})
