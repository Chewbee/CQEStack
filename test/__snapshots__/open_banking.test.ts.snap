// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stack built 1`] = `
Object {
  "Outputs": Object {
    "CommandStackrestAPIEndpoint57446DD7": Object {
      "Value": Object {
        "Fn::Join": Array [
          "",
          Array [
            "https://",
            Object {
              "Ref": "CommandStackrestAPI3679B2B9",
            },
            ".execute-api.",
            Object {
              "Ref": "AWS::Region",
            },
            ".",
            Object {
              "Ref": "AWS::URLSuffix",
            },
            "/",
            Object {
              "Ref": "CommandStackrestAPIDeploymentStageprod59D1A883",
            },
            "/",
          ],
        ],
      },
    },
  },
  "Parameters": Object {
    "AssetParameters51561b7a7253f6e2eb1ac65df257fc9bc8c85bb1f3034c3ed60b9b029315f4bcArtifactHashF72B14FE": Object {
      "Description": "Artifact hash for asset \\"51561b7a7253f6e2eb1ac65df257fc9bc8c85bb1f3034c3ed60b9b029315f4bc\\"",
      "Type": "String",
    },
    "AssetParameters51561b7a7253f6e2eb1ac65df257fc9bc8c85bb1f3034c3ed60b9b029315f4bcS3Bucket68DECA3D": Object {
      "Description": "S3 bucket for asset \\"51561b7a7253f6e2eb1ac65df257fc9bc8c85bb1f3034c3ed60b9b029315f4bc\\"",
      "Type": "String",
    },
    "AssetParameters51561b7a7253f6e2eb1ac65df257fc9bc8c85bb1f3034c3ed60b9b029315f4bcS3VersionKey88FE41EF": Object {
      "Description": "S3 key for asset version \\"51561b7a7253f6e2eb1ac65df257fc9bc8c85bb1f3034c3ed60b9b029315f4bc\\"",
      "Type": "String",
    },
  },
  "Resources": Object {
    "CommandStackCommandsTable15B01C37": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "AttributeDefinitions": Array [
          Object {
            "AttributeName": "itemId",
            "AttributeType": "S",
          },
        ],
        "KeySchema": Array [
          Object {
            "AttributeName": "itemId",
            "KeyType": "HASH",
          },
        ],
        "PointInTimeRecoverySpecification": Object {
          "PointInTimeRecoveryEnabled": true,
        },
        "ProvisionedThroughput": Object {
          "ReadCapacityUnits": 40,
          "WriteCapacityUnits": 5,
        },
        "SSESpecification": Object {
          "SSEEnabled": true,
        },
        "StreamSpecification": Object {
          "StreamViewType": "NEW_AND_OLD_IMAGES",
        },
        "TableName": "CommandsTable",
        "TimeToLiveSpecification": Object {
          "AttributeName": "ddbTtl",
          "Enabled": true,
        },
      },
      "Type": "AWS::DynamoDB::Table",
      "UpdateReplacePolicy": "Delete",
    },
    "CommandStackCommandsTableWriteScalingTargetF7B52318": Object {
      "Properties": Object {
        "MaxCapacity": 10,
        "MinCapacity": 2,
        "ResourceId": Object {
          "Fn::Join": Array [
            "",
            Array [
              "table/",
              Object {
                "Ref": "CommandStackCommandsTable15B01C37",
              },
            ],
          ],
        },
        "RoleARN": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":iam::",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":role/aws-service-role/dynamodb.application-autoscaling.amazonaws.com/AWSServiceRoleForApplicationAutoScaling_DynamoDBTable",
            ],
          ],
        },
        "ScalableDimension": "dynamodb:table:WriteCapacityUnits",
        "ServiceNamespace": "dynamodb",
      },
      "Type": "AWS::ApplicationAutoScaling::ScalableTarget",
    },
    "CommandStackUSerPoolE47D6B67": Object {
      "Properties": Object {
        "Name": "UserPoolAuthorizer",
        "ProviderARNs": Array [
          "arn:aws:cognito-idp:eu-west-1:384547244036:userpool/eu-west-1_mhbaJ4zc0",
        ],
        "RestApiId": Object {
          "Ref": "CommandStackrestAPI3679B2B9",
        },
        "Type": "COGNITO_USER_POOLS",
      },
      "Type": "AWS::ApiGateway::Authorizer",
    },
    "CommandStackputcommandC6C29518": Object {
      "DependsOn": Array [
        "CommandStackputcommandServiceRoleDefaultPolicyE7AF3FE6",
        "CommandStackputcommandServiceRole6326BD17",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParameters51561b7a7253f6e2eb1ac65df257fc9bc8c85bb1f3034c3ed60b9b029315f4bcS3Bucket68DECA3D",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters51561b7a7253f6e2eb1ac65df257fc9bc8c85bb1f3034c3ed60b9b029315f4bcS3VersionKey88FE41EF",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters51561b7a7253f6e2eb1ac65df257fc9bc8c85bb1f3034c3ed60b9b029315f4bcS3VersionKey88FE41EF",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Environment": Object {
          "Variables": Object {
            "PRIMARY_KEY": "itemId",
            "TABLE_NAME": Object {
              "Ref": "CommandStackCommandsTable15B01C37",
            },
          },
        },
        "Handler": "put_command.putCommand",
        "Role": Object {
          "Fn::GetAtt": Array [
            "CommandStackputcommandServiceRole6326BD17",
            "Arn",
          ],
        },
        "Runtime": "nodejs10.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "CommandStackputcommandServiceRole6326BD17": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "CommandStackputcommandServiceRoleDefaultPolicyE7AF3FE6": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "dynamodb:BatchGetItem",
                "dynamodb:GetRecords",
                "dynamodb:GetShardIterator",
                "dynamodb:Query",
                "dynamodb:GetItem",
                "dynamodb:Scan",
                "dynamodb:BatchWriteItem",
                "dynamodb:PutItem",
                "dynamodb:UpdateItem",
                "dynamodb:DeleteItem",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "CommandStackCommandsTable15B01C37",
                    "Arn",
                  ],
                },
                Object {
                  "Ref": "AWS::NoValue",
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "CommandStackputcommandServiceRoleDefaultPolicyE7AF3FE6",
        "Roles": Array [
          Object {
            "Ref": "CommandStackputcommandServiceRole6326BD17",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "CommandStackrestAPI3679B2B9": Object {
      "Properties": Object {
        "Description": "This service receives the command",
        "Name": "Put Command Service",
      },
      "Type": "AWS::ApiGateway::RestApi",
    },
    "CommandStackrestAPIAccount5DA4B2B2": Object {
      "DependsOn": Array [
        "CommandStackrestAPI3679B2B9",
      ],
      "Properties": Object {
        "CloudWatchRoleArn": Object {
          "Fn::GetAtt": Array [
            "CommandStackrestAPICloudWatchRoleE647DCEE",
            "Arn",
          ],
        },
      },
      "Type": "AWS::ApiGateway::Account",
    },
    "CommandStackrestAPICloudWatchRoleE647DCEE": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "apigateway.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AmazonAPIGatewayPushToCloudWatchLogs",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "CommandStackrestAPIDeployment1FEDBCF62700ad24c746a9c0558f40d57d280761": Object {
      "DependsOn": Array [
        "CommandStackrestAPIPOSTA092BBFD",
      ],
      "Properties": Object {
        "Description": "Automatically created by the RestApi construct",
        "RestApiId": Object {
          "Ref": "CommandStackrestAPI3679B2B9",
        },
      },
      "Type": "AWS::ApiGateway::Deployment",
    },
    "CommandStackrestAPIDeploymentStageprod59D1A883": Object {
      "Properties": Object {
        "DeploymentId": Object {
          "Ref": "CommandStackrestAPIDeployment1FEDBCF62700ad24c746a9c0558f40d57d280761",
        },
        "RestApiId": Object {
          "Ref": "CommandStackrestAPI3679B2B9",
        },
        "StageName": "prod",
      },
      "Type": "AWS::ApiGateway::Stage",
    },
    "CommandStackrestAPIPOSTA092BBFD": Object {
      "Properties": Object {
        "AuthorizationType": "NONE",
        "HttpMethod": "POST",
        "Integration": Object {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                Object {
                  "Ref": "AWS::Region",
                },
                ":lambda:path/2015-03-31/functions/",
                Object {
                  "Fn::GetAtt": Array [
                    "CommandStackputcommandC6C29518",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": Object {
          "Fn::GetAtt": Array [
            "CommandStackrestAPI3679B2B9",
            "RootResourceId",
          ],
        },
        "RestApiId": Object {
          "Ref": "CommandStackrestAPI3679B2B9",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "CommandStackrestAPIPOSTApiPermissionMyTestStackCommandStackrestAPIAB625E13POST1A0CAA14": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "CommandStackputcommandC6C29518",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              Object {
                "Ref": "AWS::Region",
              },
              ":",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "CommandStackrestAPI3679B2B9",
              },
              "/",
              Object {
                "Ref": "CommandStackrestAPIDeploymentStageprod59D1A883",
              },
              "/POST/",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
  },
}
`;
